os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -597,15 +583,32 @@ void up_assert(const uint8_t *filename, int lineno)
 	 */
 	if (!IS_SECURE_STATE()) {
 		print_assert_detail(filename, lineno, fault_tcb, asserted_location);
-		lldbg("Checking kernel heap for corruption...\n");
-	}
-	if (mm_check_heap_corruption(g_kmmheap) != OK) {
-		is_kmm_corruption = true;
+	} else {
+		if (mm_check_heap_corruption(g_kmmheap) != OK) {
+			/* treat kernel fault */
+			_up_assert(EXIT_FAILURE);
+		} 
 	}
 
+#if defined(CONFIG_BOARD_CRASHDUMP)
+	board_crashdump(up_getsp(), fault_tcb, (uint8_t *)filename, lineno);
+#endif
+
 	irqrestore(flags);
 
-	recovery_assert(asserted_location, is_kmm_corruption);
+	#ifdef CONFIG_BINMGR_RECOVERY
```

Comment:
Thank you. All log appears to be well organized.

Please just check the tab alignment here.

Reply:
Thank you.
Just checked the tab alignment and pushed the change.